packages/playground/src/components/app-root/app-root.tsx

```diff
@@ -234,28 +233,11 @@ export class AppRoot {
   }
 
   waitForMultisig() {
-    setTimeout(async () => {
-      const { token } = this.accountState.user;
-      const user = await PlaygroundAPIClient.getUser(token as string);
-
-      if (!user.multisigAddress) {
-        this.waitForMultisig();
-        return;
-      }
+    const { provider, user } = this.accountState;
 
-      await this.setMultisig(user.multisigAddress);
+    provider.once(user.transactionHash, async () => {
```

Comment:
Will this not race between doing a deposit and setting the multisig address on the user?

How do we ensure the multisig address is in fact set _before_ the deposit call is made?

Reply:
This doesn't race because we don't actually get the multisig address until the user clicks the "Proceed" button on the modal.

Reply:
But the following scenario is possible, right?
1. transaction gets mined so this callback is called
2. modal pops up and user clicks to proceed
3. since the transaction is mined, it's _assumed_ that the multisig has in fact been created and it's _assumed_ that the address is set on the User property, but due to some network hiccup / latency issues, it's actually not set
4. Doing a deposit then obviously fails because `multisigAddress` is undefined

This might be an edge case but it's not extremely unlikely - it could definitely be worthwhile to implement some kind of resiliency around this

Reply:
The scenario *is* possible, since we're listening events for the same action in two different places. If it's OK with you guys, I'd like to think of resiliency as an iteration (since it might be an edge case) so we can get this to `master` in a working state.

Reply:
That's fine as long as we don't forget it / put it off for too much later, as reactively dealing with state-divergence issues is usually really annoying in distributed systems and stronger guarantees (i.e. safer assumptions) are heavily preferred if their cost is relatively negligible.

Reply:
As discussed, resiliency concerns will be addressed _before_ merging this.